services/src/datasets/external/nfdi/mod.rs

```diff
@@ -0,0 +1,1031 @@
+    //     });
+    //
+    //     let res = provider.provenance(&id).await;
+    //
+    //     assert!(res.is_ok());
+    //
+    //     let res = res.unwrap();
+    //
+    //     assert!(res.provenance.is_some());
+    // }
+    //
+    // #[tokio::test]
+    // async fn it_loads_meta_data() {
+    //     let provider = new_provider().await;
+    //
+    //     let id = DatasetId::External(ExternalDatasetId {
+    //         provider_id: provider.id,
+    //         dataset_id: DATASET_ID.to_string(),
+    //     });
+    //
+    //     let _res: Box<
+    //         dyn MetaData<OgrSourceDataset, VectorResultDescriptor, VectorQueryRectangle>,
+    //     > = provider.meta_data(&id).await.unwrap();
+    //     {};
+    // }
+    //
+    // #[tokio::test]
+    // async fn it_executes_loads() {
+    //     let provider = new_provider().await;
+    //
+    //     let id = DatasetId::External(ExternalDatasetId {
+    //         provider_id: provider.id,
+    //         dataset_id: DATASET_ID.to_string(),
+    //     });
+    //
+    //     let meta: Box<
+    //         dyn MetaData<OgrSourceDataset, VectorResultDescriptor, VectorQueryRectangle>,
+    //     > = provider.meta_data(&id).await.unwrap();
+    //
+    //     let mut context = MockExecutionContext::default();
+    //     context.add_meta_data(id.clone(), meta);
+    //
+    //     let src = OgrSource {
+    //         params: OgrSourceParameters {
+    //             dataset: id,
+    //             attribute_projection: None,
+    //         },
+    //     }
+    //     .boxed();
+    //
+    //     let initialized_op = src.initialize(&context).await.unwrap();
+    //
+    //     let proc = initialized_op.query_processor().unwrap();
+    //     let proc = match proc {
+    //         TypedVectorQueryProcessor::MultiPoint(qp) => qp,
+    //         _ => panic!("Expected MultiPoint QueryProcessor"),
+    //     };
+    //
+    //     let ctx = MockQueryContext::default();
+    //
+    //     let qr = VectorQueryRectangle {
+    //         spatial_bounds: BoundingBox2D::new((-180., -90.).into(), (180., 90.).into()).unwrap(),
+    //         time_interval: TimeInterval::default(),
+    //         spatial_resolution: SpatialResolution::zero_point_one(),
+    //     };
+    //
+    //     let result: Vec<MultiPointCollection> = proc
+    //         .query(qr, &ctx)
+    //         .await
+    //         .unwrap()
+    //         .map(Result::unwrap)
+    //         .collect()
+    //         .await;
+    //
+    //     let sum: usize = result.iter().map(MultiPointCollection::len).sum();
+    //     println!("{}", sum);
+    //
+    //     let result: Vec<MultiPointCollection> = proc
+    //         .query(qr, &ctx)
+    //         .await
+    //         .unwrap()
+    //         .map(Result::unwrap)
+    //         .collect()
+    //         .await;
+    //     let sum: usize = result.iter().map(MultiPointCollection::len).sum();
+    //     println!("{}", sum);
+    // }
```

Comment:
I guess these are up to implement

Reply:
Should I remove them? We could basically reuse them after we mocked the core-storage API